# Analysis of the Gun Violence Dataset

# Part 1. &ndash; Problem Understanding & Data Understanding

Authors: Kacper Neumann, Stanisław Apanasiewicz, Jan Matusiak

----

## 1. The Dataset

### 1.1 General Information

The [Gun Violence Data](https://www.kaggle.com/datasets/jameslko/gun-violence-data) dataset contains detailed information about gun violence incidents in the United States, available in CSV format. It is based on records collected by the [Gun Violence Archive (GVA)](https://www.gunviolencearchive.org/), an independent organization created to provide a comprehensive database incidents in the U.S.

The CSV file contains data for all recorded gun violence incidents in the U.S. between January 2013 and March 2018.

In [2]:
import kagglehub
import pandas as pd
import os

path = kagglehub.dataset_download('jameslko/gun-violence-data')
df = pd.read_csv(os.path.join(path, 'gun-violence-data_01-2013_03-2018.csv'))

c:\Users\kacpe\Desktop\gun-violence-analysis\.venv\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


100%|██████████| 33.6M/33.6M [00:01<00:00, 18.5MB/s]

Extracting files...


In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 239677 entries, 0 to 239676
Data columns (total 29 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   incident_id                  239677 non-null  int64  
 1   date                         239677 non-null  str    
 2   state                        239677 non-null  str    
 3   city_or_county               239677 non-null  str    
 4   address                      223180 non-null  str    
 5   n_killed                     239677 non-null  int64  
 6   n_injured                    239677 non-null  int64  
 7   incident_url                 239677 non-null  str    
 8   source_url                   239209 non-null  str    
 9   incident_url_fields_missing  239677 non-null  bool   
 10  congressional_district       227733 non-null  float64
 11  gun_stolen                   140179 non-null  str    
 12  gun_type                     140226 non-null  str    
 13  incident_c

In [4]:
df['incident_id'].nunique()

239677

There are 239677 samples described with 29 attributes each. There are no duplicate rows.

### 1.2 Attributes

| Attribute | Type | Meaning | Unit / Format | Remarks |
|---|---|---|---|---|
| `incident_id` | Nominal | Unique identifier assigned to the incident | Positive integer ID |  |
| `date` | Ordinal | Date of the incident | Date in YYYY-MM-DD format |  |
| `state` | Nominal | U.S. state where the incident occurred | Text |  |
| `city_or_county` | Nominal | City or county of the crime | Text |  |
| `address` | Nominal | Address of the location | Text |  |
| `n_killed` | Numerical | Number of people killed | Non-negative integer count |  |
| `n_injured` | Numerical | Number of people injured | Non-negative integer count |  |
| `incident_url` | Nominal | URL to the Gun Violence Archive incident page | URL string |  |
| `source_url` | Nominal | URL to the reporting source | URL string |  |
| `incident_url_fields_missing` | Nominal | Flag indicating whether `incident_url` is missing |  |  |
| `congressional_district` | Nominal | ID of the congressional district connected to the location | Non-negative integer ID |  |
| `gun_stolen` | Nominal | Status of the guns used |  |  |
| `gun_type` | Nominal | Types of the guns used |  |  |
| `incident_characteristics` | Nominal | Details of the incident (e.g. victims, guns) |  |  |
| `latitude` | Numerical | Geographic latitude | Degrees with 4-decimal points precision (float) |  |
| `location_description` | Nominal | Location details (e.g. name of the grocery store involved) | Text |  |
| `longitude` | Numerical | Geographic longitude | Degrees with 4-decimal points precision (float) |  |
| `n_guns_involved` | Numerical | Number of guns involved | Positive integer count |  |
| `notes` | Nominal | Additional information about the crime | Text |  |
| `participant_age` | Numerical | Age of the participants at the time of the incident |  |  |
| `participant_age_group` | Ordinal | Discretized `participant_age` |  |  |
| `participant_gender` | Nominal | Gender of the participants |  |  |
| `participant_name` | Nominal | Full name of the participants |  |  |
| `participant_relationship` | Nominal | Relationship between participants (e.g. family, co-worker) |  |  |
| `participant_status` | Nominal | Status of the participants (e.g. killed, arrested) |  |  |
| `participant_type` | Nominal | Type of the participants (e.g. victim, subject-suspect) |  |  |
| `sources` | Nominal | Similar to `source_url`, may include multiple sources |  |  |
| `state_house_district` | Nominal | Voting house district | Positive integer ID |  |
| `state_senate_district` | Nominal | Voting senate district | Positive integer ID |  |

## 2. Data Mining Goals & Success Criteria

TODO